src/securityPolicy.c

```diff
@@ -499,20 +530,62 @@ security_policy_t policyForSignTxCertificateStaking(
 		ASSERT(false);
 	}
 
-	switch (txSigningMode) {
-	case SIGN_TX_SIGNINGMODE_ORDINARY_TX:
-		DENY_UNLESS(stakeCredential->type == STAKE_CREDENTIAL_KEY_PATH);
+	switch (stakeCredential->type) {
+	case STAKE_CREDENTIAL_KEY_PATH:
 		DENY_UNLESS(bip44_isOrdinaryStakingKeyPath(&stakeCredential->keyPath));
 		DENY_IF(violatesSingleAccountOrStoreIt(&stakeCredential->keyPath));
+		switch (txSigningMode) {
+		case SIGN_TX_SIGNINGMODE_ORDINARY_TX:
+		case SIGN_TX_SIGNINGMODE_PLUTUS_TX:
+			break;
+
+		case SIGN_TX_SIGNINGMODE_MULTISIG_TX:
+			DENY();
+			break;
+
+		default:
+			ASSERT(false);
+			break;
+		}
 		break;
-	case SIGN_TX_SIGNINGMODE_MULTISIG_TX:
-		DENY_UNLESS(stakeCredential->type == STAKE_CREDENTIAL_SCRIPT_HASH);
+
+	case STAKE_CREDENTIAL_SCRIPT_HASH:
+		switch (txSigningMode) {
+		case SIGN_TX_SIGNINGMODE_MULTISIG_TX:
+		case SIGN_TX_SIGNINGMODE_PLUTUS_TX:
+			break;
+
+		case SIGN_TX_SIGNINGMODE_ORDINARY_TX:
+			DENY();
+			break;
+
+		default:
+			ASSERT(false);
+			break;
+		}
 		break;
+
+	case STAKE_CREDENTIAL_KEY_HASH:
```

Comment:
I'd put KEY_HASH after KEY_PATH and before SCRIPT_HASH everywhere (security policy, ui screen or wherever it is used).

Reply:
I'd also do that, if it's not the case somewhere, that's an error.